### PR TITLE
[G7T5-38] Create preliminary UI components

### PR DESCRIPTION
**Pull request by charmjj**
# Description

Adds basic reusable UI components (navbar, searchbar) to be standardised across frontend. Included extra ui dependencies (headlessui, heroicons) as used in components stated above.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Not Applicable

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/src/components/Navbar.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+                    ) : (
+                      <Bars3Icon className="block h-6 w-6" aria-hidden="true" />
+                    )}
+                  </Disclosure.Button>
+                </div>
+                <div className="flex flex-shrink-0 items-center">
+                  <a href="/" className="text-lg font-bold leading-relaxed inline-block mr-4 py-2 whitespace-nowrap uppercase text-white">LJPS</a>
+                </div>
+                <div className="hidden md:ml-6 md:flex md:items-center md:space-x-4">
+                  {navigation.map((item) => (
```
Comment (quinncheong) [inline app/src/components/Navbar.js:53]: Maybe we can pull this and subsequent similar code blocks below out as a separate function.

File: app/src/components/Navbar.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
+import React, { Fragment } from 'react'
+import { Disclosure, Menu, Transition } from '@headlessui/react'
+import { Bars3Icon, XMarkIcon } from '@heroicons/react/24/outline'
+import { PlusIcon } from '@heroicons/react/20/solid'
+
+const navigation = [
```
Comment (quinncheong) [inline app/src/components/Navbar.js:6]: Maybe we can initialize these as react states. 

One advantage is that we can provide control flow by setting the state depending on the type of user that is logged in.

Also change name to be something more intuitive, that can better differentiate between the const below.

File: app/src/components/Navbar.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+  return false;
+}
+
+export default function Navbar() {
+  return (
+    <Disclosure as="nav" className="bg-cyan-800">
+      {({ open }) => (
+        <>
+          <div className="mx-auto max-w-7xl px-4 sm:px-6 lg:px-8">
+            <div className="flex h-16 justify-between">
```
Comment (quinncheong) [inline app/src/components/Navbar.js:36]: Looks like the div here is very nested. Do we need so many levels? Can we (1) simplify the levels , (2) refactor the divs, or (3) pull out into a function and use a loop instead.

File: app/src/components/SearchBar.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React, { Fragment } from 'react'
```
Comment (quinncheong) [inline app/src/components/SearchBar.js:1]: Remove unused imports

Comment (charmjj) [reply app/src/components/SearchBar.js]: resolved in commit e1a92a9

File: app/src/components/Navbar.js (hunk 3 above)
Comment (charmjj) [reply app/src/components/Navbar.js]: dont quite understand what you mean by (3). Removed a level in commit e1a92a9

File: app/src/components/Navbar.js (hunk 1 above)
Comment (charmjj) [reply app/src/components/Navbar.js]: done in commit e1a92a9

File: app/src/components/Navbar.js (hunk 2 above)
Comment (charmjj) [reply app/src/components/Navbar.js]: done in commit e1a92a9

**Review by quinncheong — APPROVED** [review]
LGTM. If you want can merge master into your branch first so the linting and precommit will work

**Pull request merged**